shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1874,9 +1900,33 @@ - (void)hoverEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(ios(13.4))
       break;
   }
 
-  auto packet = std::make_unique<flutter::PointerDataPacket>(1);
-  packet->SetPointerData(/*index=*/0, pointer_data);
-  [_engine.get() dispatchPointerDataPacket:std::move(packet)];
+  NSTimeInterval time = [NSProcessInfo processInfo].systemUptime;
+  BOOL isRunningOnMac = NO;
+  if (@available(iOS 14.0, *)) {
+    // This "stationary pointer" heuristic is not needed when running within macOS.
+    // We instead receive a scroll cancel event directly translated from AppKit.
+    // See gestureRecognizer:shouldReceiveEvent:
+    isRunningOnMac = [NSProcessInfo processInfo].iOSAppOnMac;
```

Comment:
why are we checking `isRunningOnMac` for `hoverEvent:`, but not in  `gestureRecognizer:shouldReceiveEvent:`

Reply:
shouldReceiveEvent is only called when running on mac. It's never called on iPad so I didn't put a check.

Reply:
Interesting, any idea why shouldReceiveEvent is only called on mac? Could you also add a comment about it?

Reply:
It is called on iOS as well but only for touches. Only on mac it has `UIEventTypeScroll`, I talked to the implementor at Apple, he said that appkit `NSEvent` is translated to `UIEvent`. On iOS, those events were never implemented in the first place.

Reply:
Doesn't iPad keyboard also support UIEventTypeScroll?

Reply:
i'm a bit confused here. Could you add comments to beginning of both functions explaining when they will be called?